src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        switch (random.nextInt(5)) {
+            case 0:
+                return new Square(color, random.nextInt(10) + 1);
+            case 1:
+                return new Rectangle(color, random.nextInt(10) + 1, random.nextInt(10) + 1);
+            case 2:
+                return new RightTriangle(color, random.nextInt(10) + 1, random.nextInt(10) + 1);
+            case 3:
+                return new Circle(color, random.nextInt(10) + 1);
+            default:
+                return new IsoscelesTrapezoid(color, random.nextInt(10) + 1, random.nextInt(10) + 1, random.nextInt(10) + 1);
```

Comment:
It's recommended to create separate variables for random values before passing them to constructors. This improves code readability. For example, `int side = random.nextInt(10) + 1;` and then use `new Square(color, side);`.